worker/httpserver/worker_test.go

```diff
@@ -308,9 +316,13 @@ attempts:
 	s.mux.ClientDone()
 
 	select {
-	case err := <-quickErr:
+	case <-quickErr:
+		// There is a race in the queueing of the request. It is possible that
+		// the timer will fire a short wait before an unheld request gets to the
+		// phase where it would return nil. However this is only under significant
+		// load, and it isn't easy to synchronise. This is why we don't actually
+		// check the error.
 		// It doesn't really matter what the error is.
-		c.Assert(err, gc.NotNil)
 	case <-time.After(coretesting.ShortWait):
```

Comment:
Should this be a LongWait as well, just in case? It's definitely in the not-expected path.